src/Interpreters/executeQuery.cpp

```diff
@@ -1100,6 +1102,11 @@ void executeQuery(
 
             {
                 auto executor = pipeline.execute();
+
+                /// For clickhouse-local to process ctrl-c in interactive mode.
+                if (set_executor_func)
```

Comment:
Maybe it will be better to provide an implementation that will return executor w/o calling execute() and use it in clickhouse-local? (too much callbacks otherwise)

Reply:
Found a better way. Now it is done via this line: https://github.com/ClickHouse/ClickHouse/blob/f6e45b31937992595e96e2fed47e1f305fe51d24/programs/local/LocalServer.cpp#L246